### DISCUSSION TITLE
Feedback On Build Issues

### DISCUSSION THREAD

**Discussion by MuhammadSulaiman001** [MuhammadSulaiman001/opengl-lab#8] [Q&A]
Please let me know if you encounter any issue when building solution's projects (makes sure you're working on a clean master branch).

**Comment by Heba-Abbas** [comment]
hello ,today I told you about a problem with cmake , here is the problem : CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at CMakeLists.txt:2 (cmake_policy):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:4 (project):
  No CMAKE_C_COMPILER could be found.



CMake Error at CMakeLists.txt:4 (project):
  No CMAKE_CXX_COMPILER could be found.



-- Configuring incomplete, errors occurred!

**Comment by MuhammadSulaiman001** [comment]
Hi @Heba-Abbas, The key issue seems to be in these lines
```
No CMAKE_C_COMPILER could be found.
No CMAKE_CXX_COMPILER could be found.
```

Please try the following...

Open Windows Start menu -> Visual Studio Installer -> modify,

1. Make sure Desktop Development With C++ is installed (checked)
2. In "Installation details" (side panel) make sure the latest "Windows 10 SDK" is installed,

Other possible solutions:

3. If you've installed other compilers (ex. MinWG, CLang), please uninstall them.
4. Uninstall CMake from control panel, download the .msi, install as adminstrator.

Let me know if that solved the issue.

**Reply by Heba-Abbas** [reply]
yup it worked thanks

**Reply by MuhammadSulaiman001** [reply]
@Heba-Abbas Glad to know that :)

Can you mention what was the core issue? or which of the suggested solutions worked for you?

**Reply by Heba-Abbas** [reply]
i didn't checked all options in installation details "it's my fault" ,so i reinstalled the environment. the first solution you suggested worked
